app/models/potential_match.rb

```diff
@@ -104,9 +104,14 @@ def self.generate_in_background(collection_id)
       required_types = settings.required_types.map {|t| t.classify}
 
       # treat each signup as a request signup first
-      collection.signups.find_each do |signup|
+      # because find_each doesn't give us a consistent order, but we don't necessarily
```

Comment:
Actually, in retrospect I realized the issue was likely because the job-restarting code was causing the process to start over and over on top of itself. However, I think this remains a better way of handling the progress (doesn't break if user changes name/pseud midstream), so would still rather get this in.